File: Problem/atcoder/ABC213/2.clj
(defn int-input [] (Long/parseLong (read-line)))
(defn int-list-input [] (map #(Long/parseLong %) (clojure.string/split (read-line) #" ")))
(defn int-row-input [n] (line-seq))

#_(defn main
    [a]
    (->> a
         (map-indexed (fn [idx item] [idx item]))
         (sort-by second >)
         second
         first
         inc))

#_(println
   (let [n (int-input)
         a (map #(bigint %) (clojure.string/split (read-line) #" "))]
     (main a)))

(println (int-list))

File: Problem/atcoder/ABC183/2.py
sg = list(map(int, input().split()))

s, g = [], []
for i in range(4):
    if i == 0 or i == 1:
        s.append(sg[i])
    else:
        g.append(sg[i])
#print(s, g)

g[1] = (-1) * g[1]

x = - s[1] * (s[0] - g[0]) / (s[1] - g[1]) + s[0]
print(x)
